dotnet/Capstone/Controllers/BirdNoteController.cs

```diff
@@ -0,0 +1,158 @@
+﻿using Capstone.DAO;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using System;
+using Capstone.Models;
+using Capstone.Exceptions;
+using System.Security.Principal;
+
+namespace Capstone.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class BirdNoteController : ControllerBase
+    {
+        private readonly IBirdNoteDao birdSightingDao;
+        public BirdNoteController(IBirdNoteDao birdSightingDao)
+        {
+            this.birdSightingDao = birdSightingDao;
+
+        }
+
+        [HttpGet("/bird/{birdId}/notes")]
+        public List<BirdNote> getNotes(int birdId)
+        {
+            List<BirdNote> sightingList = birdSightingDao.getSightings(birdId);
+
+            if (sightingList == null)
+            {
+                Console.WriteLine("No bird");
+            }
+
+            return sightingList;
+
+        }
+
+        [HttpGet("/note")]
+        public BirdNote getNote(int id)
+        {
+            BirdNote sighting = birdSightingDao.getBirdSighting(id);
+
+            if (sighting == null)
+            {
+                Console.WriteLine("No bird");
+            }
+
+            return sighting;
+
+        }
+
+        [HttpPost("/newNote/{birdId}")]
+        public IActionResult addNote([FromBody] BirdNote newBirdSighting)
+        {
+            const string errorMessage = "An error occurred and a bird was not created.";
+
+            IActionResult result;
+
+            DateTime today = DateTime.Now;
+            
```

Comment:
Where is our date validation?

Reply:
Nevermind, I see we only adjusted the editNote method. Enter this code here:
```suggestion
    int currentDate = sighting.DateSpotted.CompareTo(DateTime.Now);

        if (currentDate >= 0)
        {
            return BadRequest(new { message = "Cannot enter future date" });
        }     
```

Reply:
Done